osu.Game/Online/Leaderboards/LeaderboardManager.cs

```diff
@@ -181,7 +182,8 @@ private void localScoresChanged(IRealmCollection<ScoreInfo> sender, ChangeSet? c
 
             newScores = newScores.Detach().OrderByTotalScore();
 
-            scores.Value = LeaderboardScores.Success(newScores.ToArray(), null);
+            var newScoresArray = newScores.ToArray();
+            scores.Value = LeaderboardScores.Success(newScoresArray, newScoresArray.Length, null);
```

Comment:
The total scores field is not used in this path but a sane value is given anyway.